### DISCUSSION TITLE
`minifyIntrospectionQuery` is removing a scalar that I need

### DISCUSSION THREAD

**Discussion by bbigras** [urql-graphql/urql#1577] [Q&A] [answered]
I'm using PostGraphile, and it defines a `Datetime` in its schema for the psql `timestamptz` type.

If I use `minifyIntrospectionQuery(getIntrospectedSchema(data))`, `Datetime` is removed, but it's there if I only use `getIntrospectedSchema(data)`.

```js
import { getIntrospectionQuery } from "graphql";
import fetch from "node-fetch";
import * as fs from "fs";

import { getIntrospectedSchema, minifyIntrospectionQuery } from '@urql/introspection';

fetch("http://localhost:5000/graphql", {
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify({
    variables: {},
    query: getIntrospectionQuery(),
  }),
})
  .then((result) => result.json())
  .then(({ data }) => {
  //const minified = minifyIntrospectionQuery(getIntrospectedSchema(data));
  const minified = getIntrospectedSchema(data);
    fs.writeFile("./schema.json", JSON.stringify(minified), (err) => {
      if (err) {
        console.error("Writing failed:", err);
        return;
      }
      console.log("Schema written!");
    });
  });
```

**Comment by kitten** [comment, marked as answer]
To be very precise here, the schema that `minifyIntrospectionQuery` outputs is for use with `urql` / Graphcache only, meaning it's only meant to provide enough information for the client-side, but not any other tool as it strips out any information that Graphcache doesn't require.

**Reply by bbigras** [reply]
I'm using the schema on the client side to decode the date with `customScalarsExchange`. Is something like this a frequent use case?

```js
const scalarsExchange = customScalarsExchange({
  schema,
  scalars: {
    Datetime(value) {
          return parseISO(value);
        },
  },
});
```

**Reply by kitten** [reply]
Hm I don't really know about the `customScalarsExchange`, so I assume it's a third-party one. If it does need the scalar names, then you'll have to either leave out `minifyIntrospectionQuery` or modify it; I suppose we could actually make an option that leaves the scalar information in 🤔

**Reply by kitten** [reply]
@bbigras The next version of `minifyIntrospectionQuery` will allow you to include scalar information in your output https://github.com/FormidableLabs/urql/pull/1578
